packages/botonic-plugin-flow-builder/src/content-fields/flow-knowledge-base.tsx

```diff
@@ -11,12 +11,18 @@ export class FlowKnowledgeBase extends ContentFieldsBase {
   public text: string = ''
   public inferenceId?: string
   public sourcesData: { id: string; name: string }[] = []
+  public instructions: string
+  public hasMemory: boolean
+  public memoryLength?: number
 
   static fromHubtypeCMS(component: HtKnowledgeBaseNode): FlowKnowledgeBase {
     const newKnowledgeBase = new FlowKnowledgeBase(component.id)
     newKnowledgeBase.code = component.code
     newKnowledgeBase.feedbackEnabled = component.content.feedback_enabled
     newKnowledgeBase.sourcesData = component.content.sources_data
+    newKnowledgeBase.instructions = component.content.instructions
+    newKnowledgeBase.hasMemory = component.content.has_memory || false
+    newKnowledgeBase.memoryLength = component.content.memory_length
```

Comment:
you should consider the case when `has_memory` is false and `memory_length` is defined. I think you are not using `hasMemory` in the rest of the code.